### PR TITLE
Added data integration for mode analytics

### PR DESCRIPTION
**Pull request by mohamad-balouza**
# Summary
Added Tap for mode analytics in order to create an integration that pulls BI assets metadata ( dashboards, charts, etc. ) and load it to a destination

# Tests
Followed This [documentation](https://github.com/mage-ai/mage-ai/tree/master/mage_integrations) and made sure that the data syncs to a snowflake destination.

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_integrations/mage_integrations/sources/mode/__init__.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    def get_forced_replication_method(self, stream_id):
+        return STREAMS[stream_id].replication_method
+
+    def get_table_key_properties(self, stream_id):
+        return STREAMS[stream_id].key_properties
+
+    def get_valid_replication_keys(self, stream_id):
+        return STREAMS[stream_id].valid_replication_keys
+
+    def test_connection(self):
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/mode/__init__.py:56]: You don't need to add this method here if the test_connection logic isn't implemented.

File: mage_integrations/mage_integrations/sources/mode/schemas/reports.json [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      },
+      "query_preview": {
+        "type": ["string", "null"]
+      },
+      "view_count": {
+        "type": ["integer", "null"]
+      }
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (wangxiaoyou1993) [reply mage_integrations/mage_integrations/sources/mode/schemas/reports.json:129]: Same for other json files

**Review by wangxiaoyou1993 — COMMENTED** [review]
Could you fill in the content for README file?

**Comment by mohamad-balouza** [comment]
I made the noted changes @wangxiaoyou1993

File: mage_integrations/mage_integrations/sources/mode/__init__.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    def get_table_key_properties(self, stream_id):
+        return STREAMS[stream_id].key_properties
+
+    def get_valid_replication_keys(self, stream_id):
+        return STREAMS[stream_id].valid_replication_keys
+
+
+if __name__ == '__main__':
+    main(Mode)
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/mode/__init__.py:58]: Actually, could you add one empty line to the files?
You can see the warning here.
image

Comment (mohamad-balouza) [reply mage_integrations/mage_integrations/sources/mode/__init__.py]: only in the __init__.py file right ? if so, done.

File: mage_integrations/mage_integrations/sources/mode/__init__.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/mode/__init__.py:59]: empty line without spaces.
there're warnings about it in several files in this PR. You can see them in the PR review page.
image

Comment (mohamad-balouza) [reply mage_integrations/mage_integrations/sources/mode/__init__.py]: Oh I didn't know I could see the errors. 
It should be fixed now, sorry for the troubles.

Comment (wangxiaoyou1993) [reply mage_integrations/mage_integrations/sources/mode/__init__.py]: Thanks for fixing it!

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**